### PR TITLE
categories with ancestors, close #1070

### LINKED ISSUE #1070: filter by categories with ancestors

### PR DESCRIPTION
**Pull request by schleichardt**

### CODE REVIEW

**Comment by schleichardt** [comment]
@lauraluiz please review

some Javadoc is here:

http://schleichardt.github.io/preview-categories-with-ancetors/apidocs/io/sphere/sdk/products/search/ProductCategoriesIdTermFilterSearchModel.html

http://schleichardt.github.io/preview-categories-with-ancetors/apidocs/io/sphere/sdk/products/search/ProductCategoriesIdTermFacetSearchModel.html

File: commercetools-models/src/main/java/io/sphere/sdk/products/ProductDataProductDraftBuilderBase.java [hunk 1]
```diff
@@ -72,6 +73,20 @@ public T categories(final List<Reference<Category>> categories) {
         return categories(new LinkedHashSet<>(categories));
     }
 
+    /**
+     * Adds categories to this product draft. Alias for {@link #categories(List)} which takes the o
```
Comment (lauraluiz) [inline commercetools-models/src/main/java/io/sphere/sdk/products/ProductDataProductDraftBuilderBase.java]: I hope that's unfinished :D

Comment (schleichardt) [inline commercetools-models/src/main/java/io/sphere/sdk/products/ProductDataProductDraftBuilderBase.java]: hope?

File: commercetools-models/src/main/java/io/sphere/sdk/products/ProductDataProductDraftBuilderBase.java [hunk 2]
```diff
@@ -72,6 +73,20 @@ public T categories(final List<Reference<Category>> categories) {
...
+     *
+     * @param categories
+     * @return
+     */
+    public T categoriesObjects(final List<Category> categories) {
```
Comment (lauraluiz) [inline commercetools-models/src/main/java/io/sphere/sdk/products/ProductDataProductDraftBuilderBase.java]: About the naming of this method, I'm not sure if it's like this for some API or method or whatever reason. But if not, in English it should be `categoryObjects` (e.g. it's "milk boxes", not "milks boxes").

File: commercetools-models/src/main/java/io/sphere/sdk/products/ProductDataProductDraftBuilderBase.java (hunk 1 above)
Comment (lauraluiz) [inline commercetools-models/src/main/java/io/sphere/sdk/products/ProductDataProductDraftBuilderBase.java]: Yes, I hope the sentence doesn't finish like that.

**Comment by lauraluiz** [comment]
👍

**Pull request merged**